python/pylibcudf/pylibcudf/tests/io/test_parquet.py

```diff
@@ -107,3 +109,92 @@ def test_read_parquet_filters(
 # ^^^ This one is not tested since it's not in pyarrow/pandas, deprecate?
 # bool convert_strings_to_categories = False,
 # bool use_pandas_metadata = True
+
+
+@pytest.mark.parametrize(
+    "compression",
+    [
+        plc.io.types.CompressionType.NONE,
+        plc.io.types.CompressionType.GZIP,
+    ],
+)
+@pytest.mark.parametrize(
+    "stats_level",
+    [
+        plc.io.types.StatisticsFreq.STATISTICS_NONE,
+        plc.io.types.StatisticsFreq.STATISTICS_COLUMN,
+    ],
+)
+@pytest.mark.parametrize("int96_timestamps", [True, False])
+@pytest.mark.parametrize("write_v2_headers", [True, False])
+@pytest.mark.parametrize(
+    "dictionary_policy",
+    [
+        plc.io.types.DictionaryPolicy.ADAPTIVE,
+        plc.io.types.DictionaryPolicy.NEVER,
+    ],
+)
+@pytest.mark.parametrize("utc_timestamps", [True, False])
+@pytest.mark.parametrize("write_arrow_schema", [True, False])
+@pytest.mark.parametrize(
+    "partitions",
+    [None, [plc.io.types.PartitionInfo.from_start_and_rows(0, 10)]],
+)
+@pytest.mark.parametrize("column_chunks_file_paths", [None, ["tmp.parquet"]])
+@pytest.mark.parametrize("row_group_size_bytes", [None, 100])
+@pytest.mark.parametrize("row_group_size_rows", [None, 1])
+@pytest.mark.parametrize("max_page_size_bytes", [None, 100])
+@pytest.mark.parametrize("max_page_size_rows", [None, 1])
+@pytest.mark.parametrize("max_dictionary_size", [None, 100])
+def test_write_parquet(
```

Comment:
1. This test already generates 32k tests. Thoughts on if that's OK and if not which parameters are most important to exercise
2. So far I'm just asserting that  `assert isinstance(result, plc.io.parquet.BufferArrayFromVector)`. I'm hoping that cudf Python can assert more specifics about the metadata result (given that I'm testing a lot of params here) but happy to assert something stronger if desired

Reply:
That seems like too many. I think we have to trust that the combinations of parameters are implemented correctly by libcudf, so we can test all of these parameters in groups that make "logical" sense together.

We could also just check that round-tripping through set/get works individually for each setter.

Reply:
Sure thing. I reduced the parameters to run ~1000 tests that takes 9ish seconds on a DGX